cmake/configure_lvi_mitigation_build.cmake

```diff
@@ -1,25 +1,81 @@
 # Copyright (c) Open Enclave SDK contributors.
 # Licensed under the MIT License.
 
+macro(detect_compiler BINDIR CC)
+    # Compiler detection logic:
+    # 1. Use either the default compiler or the one specified via CC/CXX.
+    # 2. If the compiler in 1. does not exist and equals to clang/clang++,
+    #    the logic tries to pick the clang/clang++ with specific versions (order: 9, 8, 7).
+    # 3. If no compiler in 2. is found, the logic picks gcc/g++.
```

Comment:
So I would say the logic should be:

1. Use CC/CXX
2. If CC/CXX are not set, default to clang
3. If no clang, default to gcc

Reply:
I agree on that if CC is set, the logic just picks whatever it is.
However, when CC is not set, the current logic of using make is trying to use the newer version of clang, see https://github.com/openenclave/openenclave/blob/master/samples/config.mk.

The main reason is that the default clang version is old (for example, it's version 6 on ubuntu 18.04 if install via `apt install`).

Reply:
@anakrish can you add on this?

Reply:
@jhand2 I've update the PR. Now it does
1. Use CC/CXX (and fail if the specified compiler is not found).
2. If CC/CXX is not set, default to clang/clang++
3. If no clang/clang++, try clang-9/8/7 (this is what samples/config.mk does)
4. If no compiler found in 3, fallback to gcc/g++.

Reply:
Looks good to me. Thanks!